script.js

```diff
@@ -1,4 +1,350 @@
+    (playerCardTwo.name == "ace" && playerCardOne.rank >= 10)
+  ) {
+    isBlackJack = true;
+  }
+
+  return isBlackJack;
+};
+
+// Function that calculates a hand
+var calculateTotalHandValue = function (handArray) {
+  var totalHandValue = 0;
+  // Counter to keep track of the number of aces found within the given hand
+  var aceCounter = 0;
+
+  // Loop through player or dealers hand and add up the ranks
+  var index = 0;
+  while (index < handArray.length) {
+    var currCard = handArray[index];
+
+    // In blackjack, the value of king, queen, and jack are counted as 10 by default
+    if (
+      currCard.name == "king" ||
+      currCard.name == "queen" ||
+      currCard.name == "jack"
+    ) {
+      totalHandValue = totalHandValue + 10;
+    }
+    // We count the value of ace as 11 by default
+    else if (currCard.name == "ace") {
+      totalHandValue = totalHandValue + 11;
+      aceCounter = aceCounter + 1;
+      // Else, all other numbered cards are valued by their ranks
+    } else {
+      totalHandValue = totalHandValue + currCard.rank;
+    }
+    index = index + 1;
+  }
+
+  // Reset index for ace counter
+  index = 0;
+  // Loop for the number of aces found and only deduct 10 from total hand value
+  // when totalHandValue is more than 21.
+  while (index < aceCounter) {
+    if (totalHandValue > 21) {
+      totalHandValue = totalHandValue - 10;
+    }
+    index = index + 1;
+  }
+
+  return totalHandValue;
+};
+
+// Function that displays the player and dealers hand in a message
+var displayPlayerAndDealerHands = function (playerHandArray, dealerHandArray) {
+  var playerMessage = "Player hand:<br>";
+  var index = 0;
+  while (index < playerHandArray.length) {
+    playerMessage =
+      playerMessage +
+      "- " +
+      playerHandArray[index].name +
+      " of " +
+      playerHandArray[index].suit +
+      "<br>";
+    index = index + 1;
+  }
+
+  index = 0;
+  var dealerMessage = "Dealer hand:<br>";
+  while (index < dealerHandArray.length) {
+    dealerMessage =
+      dealerMessage +
+      "- " +
+      dealerHandArray[index].name +
+      " of " +
+      dealerHandArray[index].suit +
+      "<br>";
+    index = index + 1;
+  }
+
+  return playerMessage + "<br>" + dealerMessage;
+};
+
+// Function that displays the total hand values of the player and the dealer in a message
+var displayHandTotalValues = function (playerHandValue, dealerHandValue) {
+  var totalHandValueMessage =
+    "<br>Player total hand value: " +
+    playerHandValue +
+    "<br>Dealer total hand value: " +
+    dealerHandValue;
+  return totalHandValueMessage;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var outputMessage = "";
+
+  // FIRST CLICK
+  if (currentGameMode == GAME_START) {
+    // create a deck of cards
+    gameDeck = createNewDeck();
+
+    // deal 2 cards to player and dealer
+    playerHand.push(gameDeck.pop());
```

Comment:
If you see such a repetitive pattern, you can most probably refactor it into a function.
e.g.
```js
function updateHands() {
// logic here
}
```